src/pages/GrantsPage.jsx

```diff
@@ -8,18 +8,22 @@ import FilterVision from "../components/FilterVision/FilterVision";
 import FilterIncome from "../components/FilterIncome/FilterIncome";
 import FilterEnglishSL from "../components/FilterEnglishSL/FilterEnglishSL";
 import FilterDuration from "../components/FilterDuration/FilterDuration";
+import LoadingSpinner from "../components/LoadingSpinner";
 
 function GrantsPage() {
+  const [loading, setLoading] = useState(false)
   const [grantList, setGrantList] = useState([]);
   const [selectedCategory, setSelectedCategory] = useState("");
   useEffect(() => {
+    setLoading(true)
     fetch(`${process.env.REACT_APP_API_URL}scholarships/`)
       .then((results) => {
         console.log(results);
         return results.json();
       })
       .then((data) => {
         setGrantList(data);
+        setLoading(false)
       });
```

Comment:
Because you have a conditional render further down, it's also a good idea to add a catch like so:
```suggestion
  useEffect(() => {
    setLoading(true)
    fetch(`${process.env.REACT_APP_API_URL}scholarships/`)
      .then((results) => {
        console.log(results);
        return results.json();
      })
      .then((data) => {
        setGrantList(data);
        setLoading(false)
      })
      .catch(() => {
      	setLoading(false)
      })
```

This means that if your API errors out, your application isn't just stuck in limbo spinning all the time 😂

Reply:
that's a clever idea! - Appreciate the insight!